src/protocols/secure_channel/SessionEstablishmentDelegate.h

```diff
@@ -42,6 +54,16 @@ class DLL_EXPORT SessionEstablishmentDelegate
      */
     virtual void OnSessionEstablishmentError(CHIP_ERROR error) {}
 
+    /**
+     *   Called when session establishment fails with an error and state at the
+     *   failure. This will be called at most once per session establishment and
+     *   will not be called if OnSessionEstablished is called.
+     */
+    virtual void OnSessionEstablishmentError(CHIP_ERROR error, SessionEstablishmentStage stage)
```

Comment:
Followup: we should fix SetUpCodePairer.cpp and CommissioningWindowManager.cpp to call the new overload where they have direct OnSessionEstablishmentError() calls.

Reply:
https://github.com/project-chip/connectedhomeip/issues/31140